### PR TITLE
Rails documentation in sphinx

### PR DESCRIPTION
**Pull request by cilim**
PR for https://github.com/stormpath/stormpath-rails/issues/48

### CODE REVIEW

**Comment by cilim** [comment]
The pending documentation regarding ID Site authentication is now also finished and this pull request is ready to be reviewed @DamirSvrtan

File: docs/authentication.rst [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+    **HTTP Basic Authentication** or **OAuth2 Client Credentials**.
+
+
+
+Cookie Authentication
+---------------------
+
+If you are building a web application that serves traditional HTML pages, or a
+Single Page Application (Angular/React), this library will handle the cookie
+sessions for you.  No special configuration is necessary.
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: cookies for you*

File: docs/authentication.rst [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+
+To use cookie authentication, simply use the ``require_authentication!`` before callback:
+
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: before action*

File: docs/authentication.rst [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+Configuring Cookie Flags
+........................
+
+This library creates two cookies, one for the Access Token and one for the
+Refresh Token.  There are several options available for controlling the flags
+that are set on these cookies.
+
+Here is an example configuration block, with the default settings:
+
+.. code-block:: ruby
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: code-block should be yaml, not ruby.

File: docs/authentication.rst [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+|             |         | by ``req.protocol``), unless explicitly set to       |
+|             |         | ``false`` (not recommended!).                        |
++-------------+---------+------------------------------------------------------+
+
+.. _token_validation_strategy:
+
+Token Validation Strategy
+.........................
+
+When a request comes into your server, this gem will use the Access Token
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: to your server*

File: docs/authentication.rst [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+the Access Token hasn't been revoked.
+
+If you would like to check for Access Token revocation on every request, you
+should opt-in to the ``stormpath`` validation strategy.  This will make a
+network call to the Stormpath REST API.  If the Access Token has been revoked,
+or the account has been disabled or deleted, the request will be rejected.
+
+Opt-in to ``stormpath`` validation with this configuration:
+
+.. code-block:: ruby
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: same thing with code-block. Please change it in all places needed.

File: docs/authentication.rst [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+
+
+OAuth2 Client Credentials
+-------------------------
+
+If you are building an API service and you have complex needs around
+authorization and security, this strategy should be used.  In this situation
+the developer does a one-time exchange of their API Keys for an Access Token.
+This Access Token is time limited and must be periodically refreshed.  This adds a
+layer of security, at the cost of being less simple than HTTP Basic
```
Comment (DamirSvrtan) [inline docs/authentication.rst]: I would use more complex instead of less simple :)

File: docs/changelog.rst [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+Gem changes until version 2.0.2, in descending order.
+
+Version 2.0.2
+-------------
+Released on Aug 29, 2016
+- Render path links depending on the configuration
+- Use Faker to generate random test data
+- Rename all user instances to account
+
+Released on Aug 22, 2016
```
Comment (DamirSvrtan) [inline docs/changelog.rst]: Shouldn't a `Version 2.0.1` be placed on top of this?

**Comment by cilim** [comment]
Comments fixed @DamirSvrtan

**Pull request merged**